### DISCUSSION TITLE
다른 주제로 바꾸는 게 나을까요?

### DISCUSSION THREAD

**Discussion by ji006** [100-hours-a-week/discussions#76] [개발]
<실시간 주식 데이터 반영 지연 문제 해결 및 UI 개선 방안: 프론트엔드 성능 모니터링을 통한 실시간 데이터 처리 최적화 전략>을 주제로 정했는데 실시간 주식 데이터는 백엔드에서 처리해주는거라 이걸 주제로 잡는게 맞는지 모르겠습니다 그리고 실시간 주식 데이터 반영의 지연 확인 자체도 밤에만 가능해서 딥다이브 시간에 하기 어려울 것 같습니다

생각한 주제들
1. Tailwind UI와 daisyUI를 사용한 UI 컴포넌트 성능 비교 및 최적화
    * 두 라이브러리의 성능과 최적화 방법을 비교 연구 
    * UI 컴포넌트의 로딩 시간, 커스터마이징 용이성, 재사용성 등을 분석

2. Tailwind CSS의 성능 최적화 및 파일 크기 감소 기법
    * Tailwind CSS는 많은 유틸리티 클래스를 제공하기 때문에, 사용하지 않는 클래스를 제거하지 않으면 최종 CSS 파일의 크기가 커질 수 있습니다.
    * PurgeCSS나 content 옵션을 활용해 Tailwind CSS의 파일 크기를 줄이고 성능을 최적화하는 방법을 탐구
    * 최적화가 페이지 로딩 속도, Lighthouse 점수 등에 미치는 영향을 분석

3. Tailwind CSS와 전통적인 CSS 프레임워크의 성능 비교
    * Tailwind CSS와 Bootstrap, Foundation 등 전통적인 CSS 프레임워크와의 성능을 비교
    * 각 프레임워크의 CSS 파일 크기, 로딩 시간, 유지보수 용이성, 유연성 등을 주요 지표로 삼아 분석
    * 개발 속도와 코드 재사용성 측면에서 두 접근 방식의 차이점을 탐구

4. ApexCharts를 사용한 시각화 성능 최적화
    * 도넛 그래프와 섹터 그래프의 성능 최적화 방법을 연구
    * 여러 차트가 동시에 렌더링될 때의 성능, 대량의 데이터를 시각화할 때 발생하는 성능 문제, 애니메이션 효과의 영향 등을 분석
    * 사용자 인터랙션(예: 마우스 오버, 클릭 이벤트 등)과 관련된 성능 최적화 방법에 대해 탐구

5. 모의 투자 결과 시각화 최적화
    * 사용자가 모의 투자 결과를 쉽게 이해할 수 있도록 다양한 차트를 사용해 성과를 시각화하는 방법을 연구
    * 손익 그래프, 투자 비율 도넛 차트, 포트폴리오 구성 히트맵 등을 효율적으로 렌더링하고 사용자 상호작용을 최적화하는 방법을 탐구
    * 차트 라이브러리(ApexCharts, TradingView 등)를 사용하여 성능을 최적화하는 방안도 포함

**Comment by ej31** [comment]
# 답변

1. 사실 실시간 데이터 처리 최적화를 위해서는 msw 혹은 nock 같은 restapi moking library 통해서 렌더링 과정 관찰 및 관련 함수 최적화를 하는 것도 방법이긴 할 것 같습니다.
2. 어떤 주제를 하고 싶으신지 조금 더 고민해보고 후보를 알려주시면 답변드리기 수월할 것 같습니다.

**Comment by ej31** [comment]
> <실시간 주식 데이터 반영 지연 문제 해결 및 UI 개선 방안: 프론트엔드 성능 모니터링을 통한 실시간 데이터 처리 최적화 전략>을 주제로 정했는데 실시간 주식 데이터는 백엔드에서 처리해주는거라 이걸 주제로 잡는게 맞는지 모르겠습니다 그리고 실시간 주식 데이터 반영의 지연 확인 자체도 밤에만 가능해서 딥다이브 시간에 하기 어려울 것 같습니다
> 
> 생각한 주제들
> 
> 1. Tailwind UI와 daisyUI를 사용한 UI 컴포넌트 성능 비교 및 최적화
>    
>    * 두 라이브러리의 성능과 최적화 방법을 비교 연구
>    * UI 컴포넌트의 로딩 시간, 커스터마이징 용이성, 재사용성 등을 분석
> 2. Tailwind CSS의 성능 최적화 및 파일 크기 감소 기법
>    
>    * Tailwind CSS는 많은 유틸리티 클래스를 제공하기 때문에, 사용하지 않는 클래스를 제거하지 않으면 최종 CSS 파일의 크기가 커질 수 있습니다.
>    * PurgeCSS나 content 옵션을 활용해 Tailwind CSS의 파일 크기를 줄이고 성능을 최적화하는 방법을 탐구
>    * 최적화가 페이지 로딩 속도, Lighthouse 점수 등에 미치는 영향을 분석
> 3. Tailwind CSS와 전통적인 CSS 프레임워크의 성능 비교
>    
>    * Tailwind CSS와 Bootstrap, Foundation 등 전통적인 CSS 프레임워크와의 성능을 비교
>    * 각 프레임워크의 CSS 파일 크기, 로딩 시간, 유지보수 용이성, 유연성 등을 주요 지표로 삼아 분석
>    * 개발 속도와 코드 재사용성 측면에서 두 접근 방식의 차이점을 탐구
> 4. ApexCharts를 사용한 시각화 성능 최적화
>    
>    * 도넛 그래프와 섹터 그래프의 성능 최적화 방법을 연구
>    * 여러 차트가 동시에 렌더링될 때의 성능, 대량의 데이터를 시각화할 때 발생하는 성능 문제, 애니메이션 효과의 영향 등을 분석
>    * 사용자 인터랙션(예: 마우스 오버, 클릭 이벤트 등)과 관련된 성능 최적화 방법에 대해 탐구
> 5. 모의 투자 결과 시각화 최적화
>    
>    * 사용자가 모의 투자 결과를 쉽게 이해할 수 있도록 다양한 차트를 사용해 성과를 시각화하는 방법을 연구
>    * 손익 그래프, 투자 비율 도넛 차트, 포트폴리오 구성 히트맵 등을 효율적으로 렌더링하고 사용자 상호작용을 최적화하는 방법을 탐구
>    * 차트 라이브러리(ApexCharts, TradingView 등)를 사용하여 성능을 최적화하는 방안도 포함

- 우선 작성하신 내용 중 하나를 딱 골라드릴 수는 없습니다 😬
- 웬디가 보시기에 팀 프로젝트와 엮어서 고민해보면 좋을 것 같습니다. 
- 만약 제가 한다면 3번을 주제로 할 거 같긴합니다. 재밌어보이기도 하고 정량적으로 데이터가 예쁘게 떨어질거 같아서요 하핫 😊

**Reply by ji006** [reply]
감사합니다!
